src/agent/src/main.rs

```diff
@@ -394,6 +417,103 @@ async fn start_sandbox(
     Ok(())
 }
 
+// Start-up attestation-agent, CDH and api-server-rest if they are packaged in the rootfs
+fn init_attestation_components(logger: &Logger, _config: &AgentConfig) -> Result<()> {
+    // The Attestation Agent will run for the duration of the guest.
+    launch_process(
+        logger,
+        AA_PATH,
+        &vec!["--attestation_sock", AA_ATTESTATION_URI],
+        AA_ATTESTATION_SOCKET,
+        DEFAULT_LAUNCH_PROCESS_TIMEOUT,
+    )
+    .map_err(|e| anyhow!("launch_process {} failed: {:?}", AA_PATH, e))?;
+
+    let config_path = OCICRYPT_CONFIG_PATH;
+
+    // The image will need to be encrypted using a keyprovider
+    // that has the same name (at least according to the config).
+    let ocicrypt_config = serde_json::json!({
+        "key-providers": {
+            "attestation-agent":{
+                "ttrpc":CDH_SOCKET
```

Comment:
Here should be `unix:///run/confidential-containers/cdh.sock` rather than `/run/confidential-containers/cdh.sock`

Reply:
Hi @Xynnn007 , it seems that ocicrypt config isn't needed to start CDH, AA and ASR. So shouldn't be better to incorporate that change on your https://github.com/kata-containers/kata-containers/pull/9382 ?

Reply:
Yes. It'd better to add whole ocicrypt config things in that PR

Reply:
Ok, I've added a commit that removes the ocicrypt config section, so if people are happy with can fixup that